validation-test/compiler_crashers_2_fixed/0162-sr8019.swift

```diff
@@ -0,0 +1,5 @@
+// RUN: %target-typecheck-verify-swift
+
+protocol P {}
+struct A<C> {}
+extension A: P where A: P {} // expected-error {{requirement involves recursion that is not currently supported}}
```

Comment:
I don't think this particular form of tautological recursion makes sense, and I agree that that error message you quote gets to the fundamental error. I do however want to land this to stop the crashers since they come up so often, and then we can hammer on making the exact error messages perfect after that. I'll file a JIRA.

Reply:
https://bugs.swift.org/browse/SR-8053